sequence_task_1/student/include/impl.h

```diff
@@ -9,32 +9,82 @@ class StaticPlaylist
 public:
     /** @todo Member traits */
 
+    using value_type = typename Container::value_type;
+    using size_type = typename Container::size_type;
+    using difference_type = typename Container::difference_type;
+    using pointer = typename Container::pointer;
+    using const_pointer = typename Container::const_pointer;
+    using reference = typename Container::reference;
+    using const_reference = typename Container::const_reference;
+    using iterator = typename Container::iterator;
+    using const_iterator = typename Container::const_iterator;
+    using reverse_iterator = typename Container::reverse_iterator;
+    using const_reverse_iterator = typename Container::const_reverse_iterator;
+
     /** @todo Iterators */
+    const_iterator begin() const
+    {
+        return m_tracklist.cbegin();
+    }
+
+    const_iterator end() const
+    {
+        return m_tracklist.cend();
+    }
 
     StaticPlaylist() = default;
 
     /** @todo Constructor from any reversible sequence container */
+    template<class TContainer>
+    StaticPlaylist(const TContainer& container)
+        : m_tracklist(container.crbegin(), container.crend())
+    {}
 
     /** @todo Assignment from any reversible sequence container */
+    template<class TContainer>
+    StaticPlaylist& operator =(const TContainer& container)
+    {
+        m_tracklist.assign(container.crbegin(), container.crend());
+        return *this;
+    }
 
     /** @todo Add track from initializer */
     template<class... Args>
-    const Song_t& play(Args&&... songData);
+    const Song_t& play(Args&&... songData)
+    {
+        m_tracklist.emplace_back(std::forward<Args>(songData)...);
+        return m_tracklist.back();
+    }
 
     /** @todo Add track */
-    const Song_t& play(const Song_t& song);
+    const Song_t& play(const Song_t& song)
+    {
+        return *(m_tracklist.insert(m_tracklist.end(), song));
+    }
 
     /** @todo Get first track in playlist stack */
-    const Song_t& current() const;
+    const Song_t& current() const
+    {
+        return m_tracklist.back();
+    }
 
     /** @todo Skip to the next track in playlist, remove current */
-    void switchNext();
+    void switchNext()
+    {
+        m_tracklist.erase(std::prev(m_tracklist.end()));
```

Comment:
you should use pop_back for last element erasure

Reply:
Done